### PR TITLE
chore(logs): use tranlations for actions and add the case action to the reference

### LINKED ISSUE #1137: Add action type to case reference in reports
Currently only shows `Case Reference: #1`, preferably this could also show the type
`Case Reference: #1 (TIMEOUT)` or similar

### PR DESCRIPTION
**Pull request by JPBM135**
## Why?

- Add labels to case references, it's hard to know what the case referenced did without going to it, which is a pain to do repeatedly
- Refactor case system to use translations on all actions

> **Note**
>This closes #1137

<details><summary>Examples</summary>
Case log:

image

Reports:

image

</details>

### CODE REVIEW

**Comment by JPBM135** [comment]
- [ ] Ensure translations on actions

File: packages/yuudachi/src/functions/logging/generateCaseLog.ts [hunk 1]
```diff
@@ -66,16 +67,17 @@ export async function generateCaseLog(case_: Case, logChannelId: Snowflake, loca
 	}
 
 	if (case_.refId) {
-		const [reference] = await sql<[{ log_message_id: Snowflake | null }?]>`
-			select log_message_id
+		const [reference] = await sql<[{ action: CaseAction; log_message_id: Snowflake | null }?]>`
+			select log_message_id, action
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/generateCaseLog.ts:71]:
```suggestion
			select action, log_message_id
```

File: packages/yuudachi/src/functions/logging/generateCaseLog.ts [hunk 2]
```diff
@@ -66,16 +67,17 @@ export async function generateCaseLog(case_: Case, logChannelId: Snowflake, loca
...
 			from cases
 			where guild_id = ${case_.guildId}
 				and case_id = ${case_.refId}
 		`;
 
-		if (Reflect.has(reference ?? {}, "log_message_id")) {
+		if (Reflect.has(reference ?? {}, "log_message_id") && Reflect.has(reference ?? {}, "action")) {
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/generateCaseLog.ts:77]:
```suggestion
		if (Reflect.has(reference ?? {}, "action") && Reflect.has(reference ?? {}, "log_message_id")) {
```

File: packages/yuudachi/src/functions/logging/generateReportLog.ts [hunk 3]
```diff
@@ -43,19 +45,20 @@ export async function generateReportLog(report: Report, locale: string, message?
 	}
 
 	if (report.refId) {
-		const [reference] = await sql<[{ log_message_id: Snowflake | null }?]>`
-			select log_message_id
+		const [reference] = await sql<[{ action: CaseAction; log_message_id: Snowflake | null }?]>`
+			select log_message_id, action
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/generateReportLog.ts:49]:
```suggestion
			select action, log_message_id
```

File: packages/yuudachi/src/functions/logging/generateReportLog.ts [hunk 4]
```diff
@@ -43,19 +45,20 @@ export async function generateReportLog(report: Report, locale: string, message?
...
 			from cases
 			where guild_id = ${report.guildId}
 				and case_id = ${report.refId}
 		`;
 
 		const modLogChannelId = await getGuildSetting(report.guildId, SettingsKeys.ModLogChannelId);
 
-		if (modLogChannelId && Reflect.has(reference ?? {}, "log_message_id")) {
+		if (modLogChannelId && Reflect.has(reference ?? {}, "log_message_id") && Reflect.has(reference ?? {}, "action")) {
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/generateReportLog.ts:57]:
```suggestion
		if (modLogChannelId && Reflect.has(reference ?? {}, "action") && Reflect.has(reference ?? {}, "log_message_id")) {
```

File: packages/yuudachi/src/util/actionKeys.ts [hunk 5]
```diff
@@ -1,2 +1,47 @@
...
+export function caseActionLabel(key: CaseAction, locale: string, isCase = false): string {
+	switch (key) {
+		case CaseAction.Role:
+			return isCase
+				? i18next.t(`log.history.cases.action_label.restriction_case`, { lng: locale })
+				: i18next.t(`log.history.cases.action_label.restriction`, { lng: locale });
+		case CaseAction.Unrole:
+			return isCase
+				? i18next.t(`log.history.cases.action_label.unrestriction_case`, { lng: locale })
+				: i18next.t(`log.history.cases.action_label.unrestriction`, { lng: locale });
```
Comment (iCrawl) [inline packages/yuudachi/src/util/actionKeys.ts:14]: Whether you decide to inline the ternary or not, you should at least use `"` and not backticks then, if it stays like it is now.

Comment (JPBM135) [reply packages/yuudachi/src/util/actionKeys.ts:14]: Ow, true, didn't realize

Comment (JPBM135) [inline packages/yuudachi/src/util/actionKeys.ts:14]:
```suggestion
			return isCase
				? i18next.t("log.history.cases.action_label.restriction_case", { lng: locale })
				: i18next.t("log.history.cases.action_label.restriction", { lng: locale });
		case CaseAction.Unrole:
			return isCase
				? i18next.t("log.history.cases.action_label.unrestriction_case", { lng: locale })
				: i18next.t("log.history.cases.action_label.unrestriction", { lng: locale });
```

**Review by iCrawl — APPROVED** [review]

**Pull request merged**